### PR TITLE
Polish the code

### LINKED ISSUE #96: Refactor ResetPassword -> Implement a `pre-update` hook in models
PR #92 need a refactor in order to adapt to mongoose best practice using pre-hooks.

Refactor logic in `ForgotPassword.router.js` route moving the password hashing to a pre-update hook into Seekers model.

### PR DESCRIPTION
**Pull request by jesuarva**
# Description

For components in the folder `DevInfoEditz`, `DevProfile2` and `utilityComponents`:
- ADD propTypes 
- Fix linter Errors

Fixes #96 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Locally in Dev Environment 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] My changes generate no new warnings
- [x] I have NO added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: frontend/lambda-in/src/Components/utilityComponents/StateCapsule/StateCapsule.js [hunk 1]
```diff
@@ -161,6 +140,7 @@ class StateCapsule extends Component {
     const stateCapsule = { ...this.state };
     delete stateCapsule.ready;
 
+    // eslint-disable-next-line react/prop-types
```
Comment (d-vail) [inline frontend/lambda-in/src/Components/utilityComponents/StateCapsule/StateCapsule.js]: If you add children to the proptypes, this comment could be deleted.

File: frontend/lambda-in/src/Components/DevInfoEditz/DevInfoEditz.js [hunk 2]
```diff
@@ -165,36 +144,28 @@ class DevInfoEdit extends Component {
       switch (typeOfField) {
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply frontend/lambda-in/src/Components/DevInfoEditz/DevInfoEditz.js]: where?, there are no eslint-disable

[the remark this answers is not in the document]
Comment (d-vail) [reply frontend/lambda-in/src/Components/DevInfoEditz/DevInfoEditz.js]: It was `// eslint-disable-next-line comma-dangle`. For some reason I couldn't select the exact line.

File: frontend/lambda-in/src/Components/DevInfoEditz/SeekerEditUtils/BasicInfo.js [hunk 3]
```diff
@@ -76,4 +74,8 @@ const BasicInfo = (props) => {
   );
 };
 
+BasicInfo.propTypes = {
+  userInfo: PropTypes.shape({}).isRequired,
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply frontend/lambda-in/src/Components/DevInfoEditz/SeekerEditUtils/BasicInfo.js]: done

File: frontend/lambda-in/src/Components/DevLogIn/DevLogIn.js [hunk 4]
```diff
@@ -52,8 +52,8 @@ class DevLogin extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      email: '',
-      password: '',
+      email: 'reg@c14.com',
+      password: '12345678Aa$',
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply frontend/lambda-in/src/Components/DevLogIn/DevLogIn.js]: forgot to delete that.

File: frontend/lambda-in/src/Components/DevLogIn/DevLogIn.js [hunk 5]
```diff
@@ -204,41 +204,46 @@ class DevLogin extends Component {
...
-        <label htmlFor="input-submit-button">
-          <input id="input-submit-button" className={classes.submitInput} type="submit" />
-          <Button
-            className={classes.submitButton}
-            variant="contained"
-            color="primary"
-            onClick={this.handleLogin}
-          >
+    const buttonConflict =
+      getGS('login') !== 'conflict' ? (
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply frontend/lambda-in/src/Components/DevLogIn/DevLogIn.js]: fixed

File: frontend/lambda-in/src/Components/utilityComponents/DropDown/DropDown.js [hunk 6]
```diff
@@ -1,38 +1,40 @@
...
-            <ExpansionPanelDetails>
-              <div className="expansionPanel">{this.props.children}</div>
-            </ExpansionPanelDetails>
-          </ExpansionPanel>
-        </div>
-      </Fragment>
-    );
-  }
-}
+// eslint-disable-next-line react/prop-types
```
[the remark this answers is not in the document]
Comment (jesuarva) [reply frontend/lambda-in/src/Components/utilityComponents/DropDown/DropDown.js]: done

**Comment by d-vail** [comment]
Can you explain your reasoning for describing some of the proptype shapes in a different file? Declaring them in seekerPropTypes.js? I understand that you've saved a little bit of time not having to type that up in each component but it makes it more difficult to understand what props a component is using. Now if someone wants the full picture they have to check the proptypes on the component and look for and check the seekerPropTypes.js file.

**Comment by jesuarva** [comment]
Hello Danny,

Currently, the propTypes for the project are no static props, they are susceptible to be modified (as an example the props for geolocation).

Moving them to a separate file Is not about saving some time typing (or copy-pasting), is more about being productive and having a code that is more maintainable.

For more general purpose components I agree with you that PropTypes shall be described in the same file; but our components are architectured tailoring our APP needs, few of them are utilizable in other projects without a big refactor.

**Comment by d-vail** [comment]
Readability goes hand in hand with productivity. And if we change our data models, we are still going to have to go through files fixing components. I don't agree that it is worth the cost in readability.

**Pull request merged**